src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -56,12 +59,20 @@ internal class GrpcWorker : IWorker
             _startupOptions = startupOptions ?? throw new ArgumentNullException(nameof(startupOptions));
             _serializer = workerOptions.Value.Serializer ?? throw new InvalidOperationException(nameof(workerOptions.Value.Serializer));
             _inputConversionFeatureProvider = inputConversionFeatureProvider ?? throw new ArgumentNullException(nameof(inputConversionFeatureProvider));
+            _functionMetadataProvider = functionMetadataProvider ?? throw new ArgumentNullException(nameof(functionMetadataProvider));
         }
 
         public async Task StartAsync(CancellationToken token)
         {
             var eventStream = _rpcClient.EventStream(cancellationToken: token);
 
+            if(_startupOptions.Value.ScriptRoot is null)
+            {
+                throw new ArgumentNullException(nameof(_startupOptions.Value.ScriptRoot));
+            }
+
+            _functionMetadataResponseTask = GetFunctionMetadataAsync(_startupOptions.Value.ScriptRoot);
```

Comment:
Should we check workerIndexing is enabled before starting this task?

Reply:
From the last meetings I've had, the plan for the isolated worker moving forward is going to be 100% worker-indexing. Technically, we already have that with the SDK (we load functions to the host through an extension, and host doesn't actually index functions). 

This PR basically guts that extension and moves that functionality into the worker in preparation towards source generation. We're not checking for `workerIndexing` to be enabled because there's no fallback to host-indexing for this worker (unlike other OOP workers). 

cc: @fabiocav